Managed data table (#1416)
commercetools/ui-kit: packages/components/managed-data-table/README.md

> Please be aware that it may be subject to upcoming breaking changes as it's still in active development.

---

## Description

Allows to have column manager and density manager panels to facilitate the usage of the `<DataTable />` component.

- The `disableDisplaySettings` enables / disables the display settings panel, allowing the user to select wrapping text and density display options.
- The `disableColumnManager` enables / disables the column manager panel, allowing the user to select which columns are visible.

Both panels delegate the handling of the settings change on the parent through function properties, allowing the settings to be persisted or just used as state props.

## Usage

```jsx
import ManagedDataTable from '@commercetools-uikit/managed-data-table';

const rows = [
  { id: 'parasite', title: 'Parasite', country: 'South Korea' },
  { id: 'portrait', title: 'Portrait of a Lady on Fire', country: 'France' },
  { id: 'wat', title: 'Woman at War', country: 'Iceland' },
];

const columns = [
  { key: 'title', label: 'Title' },
  { key: 'country', label: 'Country' },
];

return <ManagedDataTable rows={rows} columns={columns} />;
```

> ## Properties

Author: Maybe a bit of a silly question but I thought you couldn't set relative links within the storybook. How can I actually set the link to the DataTable readme? Should I use https://uikit.commercetools.com/?path=/story/components-table-new--datatable ?

> ## Properties

Reviewer: I would use the full link from github perspective, because the readme is used both in github and npm.
I wouldn't bother for storybook

> # DataTable

Reviewer: ```suggestion
# ManagedDataTable
```

> THIS COMPONENT IS IN BETA!

> > This component is ultimately intented to replace the current `Table` component using the new `DataTable` with display settings and column manager.

Reviewer 2: Nit - typo: `intented` -> `intended`

> Please be aware that it may be subject to upcoming breaking changes as it's still in active development.

---

> ## Usage
> 
> ```js
> import { ManagedDataTable } from '@commercetools-uikit/managed-data-table';
> ```
> 
> ### `<ManagedDataTable />`
> 
> Allows to have column manager and density manager panels to facilitate the usability of the `<DataTable />` component.

Reviewer: ```suggestion
## Description

Allows to have column manager and density manager panels to facilitate the usage of the `<DataTable />` component.
```

- The `disableDisplaySettings` enables / disables the display settings panel, allowing the user to select wrapping text and density display options.
- The `disableColumnManager` enables / disables the column manager panel, allowing the user to select which columns are visible.

Both panels delegate the handling of the settings change on the parent through function properties, allowing the settings to be persisted or just used as state props.

## Usage

> ```jsx
> const rows = [

Reviewer: ```suggestion
```jsx
import ManagedDataTable from '@commercetools-uikit/managed-data-table';

const rows = [
```

| `renderItem`           | `func`    | -        | -       | Callback function to render the content of cells under this column, overriding the default `itemRenderer` prop of the table. The `row` object and `isRowCollapsed` state are passed as arguments to this fn.                                                                                                                                                                                                                                                                                    |
| `isTruncated`          | `boolean` | -        | `false` | Set this to `true` to allow text content of this cell to be truncated with an ellipsis, instead of breaking into multiple lines. Read the note about this option below.                                                                                                                                                                                                                                                                                               |

> | `isSortable`           | `boolean` | -        | `false` | If this is `true`, this column's header will be a become a button which calls `onSortChange` when clicked. You should enable this flag for every column you want to be able to sort. When at least one column is sortable, the table props `sortBy`, `sortDirection` and `onSortChange` should be set.                                                                                                                                                                |

Reviewer 2: Nit - typo: `will be a become a button` -> `will become a button`

# ManagedDataTable

> THIS COMPONENT IS IN BETA!

> > This component is ultimately intended to replace the current `Table` component using the new `DataTable` with display settings and column manager.

Reviewer: Remove this sentence 

```suggestion
```

> Please be aware that it may be subject to upcoming breaking changes as it's still in active development.

---

> ## Description
> 
> Allows to have column manager and density manager panels to facilitate the usage of the `<DataTable />` component.

Reviewer: ```suggestion
## Description

This component enhances the `<DataTable>` component and additionally provides a UI and state management to handle configuration of the table such as column manager.
```

> # Table

Reviewer 3: In this Readme I would rather just document the features added by `ManagedTable` in relation to the simpler `DataTable`, without duplicating the documentation of the DataTable but referring to its documentation instead (as well as the other components that are part of ManagedTable, such as the ColumnManager).
This would avoid having to duplicate every change to the docs of DataTable to this file, which can easily go out of hand.
